### ISSUE TITLE
<Plug>(highlightedundo-undo/redo) overwrites Register 0

### ISSUE THREAD

**Issue by ratheesh** [machakann/vim-highlightedundo#3]
Using `<Plug>(highlightedundo-undo/redo)` overwrites Register 0

Following is my minimal .vimrc
```vim
set runtimepath+=~/.vim
call plug#begin('~/.vim/plugged')
    Plug 'machakann/vim-highlightedundo'
call plug#end()
filetype plugin indent on
filetype plugin on

let g:highlightedundo#highlight_mode            = 2
let g:highlightedundo#highlight_duration_delete = 200
let g:highlightedundo#highlight_duration_add    = 250
nmap u     <Plug>(highlightedundo-undo)
nmap <C-r> <Plug>(highlightedundo-redo)
nmap U     <Plug>(highlightedundo-Undo)
nmap g-    <Plug>(highlightedundo-gminus)
nmap g+    <Plug>(highlightedundo-gplus)
```
I did not check with other commands.
My test was done on the latest vim and neovim builds.

**Comment by machakann** [comment]
I tried the following:

1. `:new<CR>` to open an empty buffer
1. `:call setline(1, ['foo', 'bar'])<CR>` to fill buffer
1. `dd` to delete `foo`
1. `yy` to yank `bar`
1. `:echo @0<CR>` to check the register 0, it outputs `bar`
1. `u` to undo deleting `foo`
1. `:echo @0<CR>`

After the last step the register 0 was still `bar`. I may miss something, could you provide the steps to reproduce?

**Comment by ratheesh** [comment]
I followed your steps, it worked fine when I repeated exactly as mentioned in your comment. But I continued to issue `dd` on foo followed by `u` for the second time, after which register 0 was overwritten with undo content.

Following is the sequence I tried with neovim

`:new<CR>` to open an empty buffer
`:call setline(1, ['foo', 'bar'])<CR>` to fill buffer
`dd` to delete `foo`
`yy` to yank `bar`
`:echo @0<CR>` to check the register 0, it outputs `bar`
`u` to undo deleting `foo`
`:echo @0<CR>`  it outputs `bar` as expected
                           .... continue following commands without quitting vim session
`dd` to delete `foo`
`:echo @0<CR>` to check that register 0, it outputs `bar`
`u` to undo deleting `foo`
`:echo @0<CR>`   it outputs `foo` which is the undo content


I repeated this without plugin loaded several times and it works as expected.
